pallets/frequency-tx-payment/src/lib.rs

```diff
@@ -372,6 +372,57 @@ where
 		}
 	}
 
+	fn can_withdraw_fee(
+		&self,
+		who: &T::AccountId,
+		call: &<T as frame_system::Config>::RuntimeCall,
+		info: &DispatchInfoOf<<T as frame_system::Config>::RuntimeCall>,
+		len: usize,
+	) -> Result<BalanceOf<T>, TransactionValidityError> {
+		match call.is_sub_type() {
+			Some(Call::pay_with_capacity { call }) =>
+				self.can_withdraw_capacity_fee(who, &vec![*call.clone()], len),
+
+			Some(Call::pay_with_capacity_batch_all { calls }) =>
+				self.can_withdraw_capacity_fee(who, calls, len),
+
+			_ => self.can_withdraw_token_fee(who, call, info, len, self.tip(call)),
+		}
+	}
+
+	fn can_withdraw_capacity_fee(
```

Comment:
Minor: I expected this to return a boolean, but it returns a value instead. A name like try_withdraw_capacity_fee might better reflect its behavior.